### PR TITLE
fix binomial sampling

### PR DESCRIPTION
**Pull request by DavideScassola**
The logits parameter of `distributions.Binomial` is not the log of the probabilities, it's actually the log-odds. So here I just fix the problem by using the probs parameter instead.

### CODE REVIEW

File: cirkit/backend/torch/layers/input.py [hunk 1]
```diff
@@ -534,8 +534,10 @@ def log_partition_function(self) -> Tensor:
         return torch.zeros(size=(self.num_folds, 1, self.num_output_units), device=device)
 
     def sample(self, num_samples: int = 1) -> Tensor:
-        logits = torch.log(self.probs()) if self.logits is None else self.logits()
-        dist = distributions.Binomial(self.total_count, logits=logits)
+        probs = self.probs()
+        if probs is None:
+            probs = torch.exp(self.logits())
```
Comment (adrianjav) [inline cirkit/backend/torch/layers/input.py:539]: I don't think this solves the issue, right? 

Now, if logits is specified it is being used as $\log p$ in the sampling, but as $\log \frac{p}{1-p}$ in other parts of the code: https://github.com/april-tools/cirkit/blob/64d7606eab4ce94643c72b20df8f4e2f7980e3cf/cirkit/backend/torch/layers/input.py#L525

I'd advocate for using logits as the log-odds everywhere, that is, doing logits=logits in the sampling as well.

Comment (adrianjav) [reply cirkit/backend/torch/layers/input.py:539]: Isn't the line I just pointed out using it as logits?

Comment (adrianjav) [reply cirkit/backend/torch/layers/input.py:539]: Thanks! As I said, I'd personally keep them as logits and change `sample` (otherwise I'd rename the variable `logits` to `logp` or something on that line)

File: cirkit/backend/torch/layers/input.py [hunk 2]
```diff
@@ -534,8 +534,13 @@ def log_partition_function(self) -> Tensor:
...
     def sample(self, num_samples: int = 1) -> Tensor:
-        logits = torch.log(self.probs()) if self.logits is None else self.logits()
-        dist = distributions.Binomial(self.total_count, logits=logits)
+        if self.logits is not None:
+            logits = self.logits()  # (F, 1, K)
+            dist = distributions.Binomial(self.total_count, logits=logits)
+        else:
+            probs = self.probs()  # (F, 1, K)
+            dist = distributions.Binomial(self.total_count, probs=probs)
+        dist = distributions.Binomial(self.total_count, probs=probs)
```
Comment (adrianjav) [inline cirkit/backend/torch/layers/input.py:543]: remove this last line and I think we are good to go :)
